### DISCUSSION TITLE
[FEATURE] Integrate Mpesa as a gateway, and other components

### DISCUSSION THREAD

**Discussion by compgeniuses** [juspay/hyperswitch#3335] [Ideas / Feature Requests]
Implementation of  [Mpesa ](https://en.wikipedia.org/wiki/M-Pesa) as a Possible Gateway.

https://www.safaricom.co.ke/personal/m-pesa

Api docs: https://developer.safaricom.co.ke/
Mpesa is one of the largest Payment processor on the African Continent, Mostly used In kenya for sending Money, MAking payments to Merchant's, and disbursements of Funds

Possible Implementation
Possible Features for Implementation would be:

Push STK for Lipa na Mpesa (MErchant Till)
Push STK for PAybill (Account based merchant accounts)
IPN Support, Instant Payment Notification for Payments Made to a merchant till or Paybill account
Bulk Payment, for sending/ Release of Money in bulk to end Users, Suppliers or others.
Send to Bank Payments

This addition will Open up hyperswitch for use and integrations accross East Africa, as well as Africa.

_Maintainer edit: Corresponding feature request: #3334_

**Comment by manojradhakrishnan** [comment]
Thanks for the suggestion and details on the use cases.

We are really fascinated by the success of Mpesa in African continent. But our bandwidth have been dedicated for North Americas and European Union as the target geographies for Q1 2024

While this is being said, we would be happy to accept or support any code contributors to add Mpesa in Hyperswitch and expand the coverage.

**Comment by compgeniuses** [comment]
if i could properly understand Hy the hyperswitch integrations i could have someone form my team implement it, as we have vast expreince integrating Mpesa into platforms.

**Comment by Narayanbhat166** [comment]
Yes @compgeniuses, the following documentation can be followed to integrate a new connector https://github.com/juspay/hyperswitch/blob/main/add_connector.md
